indico/modules/events/editing/operations.py

```diff
@@ -260,47 +308,56 @@ def delete_revision_comment(comment):
     comment.is_deleted = True
     db.session.flush()
     logger.info('Comment on revision %r deleted: %r', comment.revision, comment)
+    comment.revision.editable.log(EventLogRealm.reviewing, LogKind.negative, 'Editing',
+                                  f'Comment on {comment.revision.editable.log_title} deleted',
+                                  session.user)
 
 
 def create_new_tag(event, code, title, color, system=False):
     tag = EditingTag(code=code, title=title, color=color, system=system, event=event)
     db.session.flush()
     logger.info('Tag %r created by %r', tag, session.user)
+    tag.log(EventLogRealm.management, LogKind.positive, 'Editing', f'Tag {tag.verbose_title} created', session.user)
     return tag
 
 
-def update_tag(tag, code=None, title=None, color=None):
-    if code:
-        tag.code = code
-    if title:
-        tag.title = title
-    if color:
-        tag.color = color
+def update_tag(tag, updates):
+    changes = tag.populate_from_dict({k: v for k, v in updates.items() if v is not None})
     db.session.flush()
     logger.info('Tag %r updated by %r', tag, session.user)
+    log_fields = {'code': 'Code', 'title': 'Title', 'color': 'Color'}
+    tag.log(EventLogRealm.management, LogKind.change, 'Editing', f'Tag {tag.verbose_title} updated', session.user,
+            data={'Changes': make_diff_log(changes, log_fields)})
 
 
 def delete_tag(tag):
     logger.info('Tag %r deleted by %r', tag, session.user)
     db.session.delete(tag)
+    tag.log(EventLogRealm.management, LogKind.negative, 'Editing', f'Tag {tag.verbose_title} deleted', session.user)
```

Comment:
Probably safer to do it the other way round. SA shouldn't be clearing anything in the deleted model, but it feels a bit dirty to call something on a deleted object...

```suggestion
    tag.log(EventLogRealm.management, LogKind.negative, 'Editing', f'Tag {tag.verbose_title} deleted', session.user)
    db.session.delete(tag)
```